core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -135,24 +139,31 @@ public WorkerNetAddress getFieldValue(MasterWorkerInfo o) {
    *
    * The block master uses concurrent data structures to allow non-conflicting concurrent access.
    * This means each piece of metadata should be locked individually. There are two types of
-   * metadata in the {@link DefaultBlockMaster}; {@link MasterBlockInfo} and
-   * {@link MasterWorkerInfo}.
-   * Individual objects must be locked before modifying the object, or reading a modifiable field
-   * of an object. This will protect the internal integrity of the metadata object.
+   * metadata in the {@link DefaultBlockMaster}; block metadata and worker metadata.
    *
-   * Lock ordering must be preserved in order to prevent deadlock. If both a worker and block
-   * metadata must be locked at the same time, the worker metadata ({@link MasterWorkerInfo})
-   * must be locked before the block metadata ({@link MasterBlockInfo}).
+   * To modify or read a modifiable piece of worker metadata, the {@link MasterWorkerInfo} for the
+   * worker must be locked. For block metadata, the id of the block must be locked. This will
+   * protect the internal integrity of the block and worker metadata.
+   *
+   * Lock ordering must be preserved in order to prevent deadlock. If both worker and block
+   * metadata must be locked at the same time, the worker metadata must be locked before the block
+   * metadata
    *
    * It should not be the case that multiple worker metadata must be locked at the same time, or
-   * multiple block metadata must be locked at the same time. Operations involving different
-   * workers or different blocks should be able to be performed independently.
+   * multiple block metadata must be locked at the same time. Operations involving different workers
+   * or different blocks should be able to be performed independently.
+   */
+
+  /**
+   * 10k locks balances between keeping a small memory footprint and avoiding unnecessary lock
+   * contention. Each stripe is around 100 bytes, so this takes about 1MB. Block locking critical
+   * sections are short, so it is acceptable to occasionally have conflicts where two different
+   * blocks want to lock the same stripe.
    */
+  private final Striped<Lock> mBlockLocks = Striped.lock(10_000);
```

Comment:
Is this a new style for larger numbers?

Reply:
https://docs.oracle.com/javase/7/docs/technotes/guides/language/underscores-literals.html

Reply:
Right, do we need to add a checkstyle or are we planning to apply this at the author's discretion?

Reply:
I think it's a nice to have but very low priority

Reply:
Yeah, if its not a simple line in the style checker I'm fine with leaving it up to the author.